tests/Unit/DataStructures/Test_VectorAlgebra.cpp

```diff
@@ -0,0 +1,74 @@
+// Distributed under the MIT License.
+// See LICENSE.txt for details.
+
+#include "tests/Unit/TestingFramework.hpp"
+
+#include <cstddef>
+#include <random>
+
+#include "DataStructures/ComplexDataVector.hpp"
+#include "DataStructures/DataVector.hpp"
+#include "DataStructures/ModalVector.hpp"
+#include "DataStructures/VectorAlgebra.hpp"
+#include "Utilities/Gsl.hpp"
+#include "Utilities/TypeTraits.hpp"
+#include "tests/Unit/TestHelpers.hpp"
+#include "tests/Utilities/MakeWithRandomValues.hpp"
+
+template <typename RhsVectorType, typename LhsVectorType>
+void test_outer() {
+  MAKE_GENERATOR(gen);
+  UniformCustomDistribution<size_t> size_distribution{5, 10};
+  UniformCustomDistribution<
+      tt::get_fundamental_type_t<get_vector_element_type_t<LhsVectorType>>>
+      lhs_distribution{-2.0, 2.0};
+  UniformCustomDistribution<
+      tt::get_fundamental_type_t<get_vector_element_type_t<RhsVectorType>>>
+      rhs_distribution{-2.0, 2.0};
+  const LhsVectorType lhs = make_with_random_values<LhsVectorType>(
+      make_not_null(&gen), make_not_null(&lhs_distribution),
+      size_distribution(gen));
+  const RhsVectorType rhs = make_with_random_values<RhsVectorType>(
+      make_not_null(&gen), make_not_null(&rhs_distribution),
+      size_distribution(gen));
+  const auto outer_product = outer(lhs, rhs);
+  size_t product_index = 0;
+  for (const auto& b : rhs) {
+    for (const auto& a : lhs) {
+      // Iterable approx consistently works for all contained types
```

Comment:
I'm not sure what this is trying to tell me. I suggest rephrasing or removing.

Reply:
I think this is trying say one of:
- `iterable_approx` recursive goes down the container hierarchy until it reaches fundamental types
- It works each time you call it in the same manner as it did previous times it was called
- It can handle different fundamental types that can be compared in a consistent manner.
Maybe something else :man_shrugging:. Is there an ambiguity here that requires documenting the behavior of `CHECK_ITERABLE_APPROX` here as opposed to what is in the `CHECK_ITERABLE_APPROX` documentation?

Reply:
It's probably not enough of a subtlety to justify the comment. The main point is just that `CHECK(approx(a) == b)` doesn't work for complex types so it's more convenient to use something with overloads.

Reply:
Ah, okay. Well I think basically what you just said would be fine:
```cpp
// Because CHECK(approx(a) == b) doesn't work for complex types, while CHECK_ITERABLE_APPROX does
```
I can see that making someone's life easier in the future :)